classes/Pods.php

```diff
@@ -1103,51 +1036,111 @@ public function field( $name, $single = null, $raw = false ) {
 
 					$object_field_found = true;
 				}
-			} elseif ( 0 === strpos( $params->name, 'image_attachment.' ) ) {
-				$size = 'thumbnail';
-
-				$image_id = 0;
-
-				$field_names = explode( '.', $params->name );
+			}
 
-				if ( isset( $field_names[1] ) ) {
-					$image_id = $field_names[1];
-				}
+			if ( ! $object_field_found ) {
+				foreach ( $image_fields as $image_field ) {
+					if ( isset( $this->fields[ $image_field ] ) ) {
+						// Skip any names that are registered fields within this Pod.
+						continue;
+					}
+					if (
+						$image_field === $params->name ||
+						0 === strpos( $params->name, $image_field . '.' )
+					) {
+						// Is it a URL request?
+						$url = '_url' === substr( $image_field, -4 );
+						if ( $url ) {
+							$image_field = substr( $image_field, 0, -4 );
+						}
 
-				if ( isset( $field_names[2] ) ) {
-					$size = $field_names[2];
-				}
+						// Results in an empty array if no traversal names are passed.
+						$traverse_names = explode( '.', $params->name );
+						array_shift( $traverse_names );
+
+						$attachment_id = 0;
+						switch ( $image_field ) {
+							case 'post_thumbnail':
+								$attachment_id = get_post_thumbnail_id();
+								break;
+							case 'image_attachment':
+								if ( isset ( $traverse_names[0] ) ) {
+									$attachment_id = $traverse_names[0];
+									array_shift( $traverse_names );
+								}
+								break;
+						}
 
-				if ( ! empty( $image_id ) ) {
-					$value = pods_image( $image_id, $size, 0, null, true );
+						if ( $attachment_id ) {
+
+							$size = 'thumbnail';
+							if ( isset( $traverse_names[0] ) ) {
+								$size  = $traverse_names[0];
+								$sizes = get_intermediate_image_sizes();
+								// Not shown by default.
+								$sizes[] = 'full';
+								$sizes[] = 'original';
+								if ( ! in_array( $size, $sizes, true ) ) {
+									// No valid image size found.
+									$size = false;
+								}
+							}
 
-					if ( ! empty( $value ) ) {
-						$object_field_found = true;
-					}
-				}
-			} elseif ( 0 === strpos( $params->name, 'image_attachment_url.' ) ) {
-				$size = 'thumbnail';
+							if ( $url ) {
+								$value              = pods_image_url( $attachment_id, $size, 0, true );
+								$object_field_found = true;
+								break;
+							}
 
-				$image_id = 0;
+							if ( $size ) {
+								$value = pods_image( $attachment_id, $size, 0, null, true );
```

Comment:
:warning: Equals sign not aligned with surrounding assignments; expected 14 spaces but found 1 space

<sub>hash: 9f6430965896fda6f73956250d736c43</sub>